src/capture/macos/appkit.rs

```diff
@@ -76,40 +84,24 @@ unsafe fn ns_string_to_string(ns_string: *mut Object) -> Option<String> {
     Some(string)
 }
 
-pub fn get_frontmost_app_pid() -> Option<i32> {
-    unsafe { 
-        let shared_workspace: *mut Object = msg_send![class!(NSWorkspace), sharedWorkspace];
-        
-        let frontmost_application: *mut Object = msg_send![shared_workspace, frontmostApplication];
-
-        if frontmost_application.is_null() {
-            return None;
-        }
-
-        let pid: i32 = msg_send![frontmost_application, processIdentifier];
-
-        if pid == -1 {
-            return None;
-        }
-
-        Some(pid)
-    }
-}
-
 /// Gets all currently running apps that may have UIs and are visible in the dock.
 /// Reference: https://developer.apple.com/documentation/appkit/nsapplicationactivationpolicy?language=objc
-pub fn get_running_apps() -> Vec<MacOSApp> {
-    let mut vec = vec![];
+pub fn get_windows() -> (Option<i32>, Vec<MacOSWindow>) {
+    let mut focused_window_id: Option<i32> = None;
+
+    let mut windows: Vec<MacOSWindow> = vec![];
+
+    let mut process_data_map: FxHashMap<i32, Arc<MacOSProcessData>> = FxHashMap::default();
```

Comment:
Using `FxHashMap`, because we don't need cryptographically secure hashing algorithm in this use-case.

Using `Arc<MacOSProcessData>`, because there's no need to clone the same process data for each window that process has.

Reply:
I doubt it's worth adding a dependency just to make this small hash map a tiny bit faster, but I guess it's fine